src/cider/exec/nextgen/jitlib/llvmjit/LLVMJITFunction.cpp

```diff
@@ -254,7 +255,11 @@ void LLVMJITFunction::cloneFunctionRecursive(llvm::Function* fn) {
   for (auto it = llvm::inst_begin(fn), e = llvm::inst_end(fn); it != e; ++it) {
     if (llvm::isa<llvm::CallInst>(*it)) {
       auto& call = llvm::cast<llvm::CallInst>(*it);
-      cloneFunctionRecursive(call.getCalledFunction());
+      llvm::Function* called_func = call.getCalledFunction();
+      if (called_func && called_func->hasFnAttribute(llvm::Attribute::AlwaysInline)) {
+        // Only symbol-called function with always-inline atrribute needs to clone.
+        cloneFunctionRecursive(called_func);
+      }
     }
```

Comment:
what determines the `AlwaysInline` attribute (eg, func definition) ? And no need to clone if not go through this branch ?

Reply:
seems AlwaysInline equals recursion termination condition？

Reply:
Runtime function defined with ```ALWAYSINLINE``` has ```AlwaysInline``` attribute. The only reason to clone function definition is inlining the function, or function declaration is enough.

Reply:
So for the runtime function which called object's API should not use `ALWAYS_INLINE` label? like call `ordered_set::insert()`.

Reply:
> So for the runtime function which called object's API should not use `ALWAYS_INLINE` label? like call `ordered_set::insert()`.

No such restriction.